File: frontend/src/types/copyrightGuide.ts
/** 版权登记指南类型定义 */

export interface GuideStep {
  step: number
  title: string
  description: string
  required_files: string[]
}

export interface RegistrationGuide {
  id: string
  work_type: string
  title_zh: string
  steps: GuideStep[]
  estimated_days: number
  estimated_fee_yuan: number
}

export interface CopyrightRegistration {
  id: string
  title: string
  work_type: string
  registration_type: string
  status: string
  application_number?: string
  registration_date?: string
  fee_yuan?: number
  notes?: string
  created_at: string
  updated_at: string
}

export interface RegistrationSummary {
  total: number
  by_status: Record<string, number>
  by_type: Record<string, number>
  total_fees_yuan: number
}

File: backend/app/schemas/copyright_guide.py
"""版权登记指南 Pydantic schemas."""

from pydantic import BaseModel
from typing import Optional


class RegistrationCreate(BaseModel):
    title: str
    work_type: str
    registration_type: str = "domestic"


class RegistrationUpdate(BaseModel):
    status: Optional[str] = None
    application_number: Optional[str] = None
    fee_yuan: Optional[float] = None
    notes: Optional[str] = None
    certificate_url: Optional[str] = None


class RegistrationResponse(BaseModel):
    id: str
    title: str
    work_type: str
    registration_type: str
    status: str
    application_number: Optional[str] = None
    registration_date: Optional[str] = None
    fee_yuan: Optional[float] = None
    notes: Optional[str] = None
    created_at: str
    updated_at: str


class GuideStep(BaseModel):
    step: int
    title: str
    description: str
    required_files: list[str]


class RegistrationGuide(BaseModel):
    id: str
    work_type: str
    title_zh: str
    steps: list[GuideStep]
    estimated_days: int
    estimated_fee_yuan: float


class RegistrationSummary(BaseModel):
    total: int
    by_status: dict
    by_type: dict
    total_fees_yuan: float
